File: app/src/main/java/com/tronography/pixelweather/dagger/SharedPrefModule.java
package com.tronography.pixelweather.dagger;

import android.content.Context;
import android.support.annotation.NonNull;

import com.tronography.pixelweather.utils.SharedPrefsUtils;

import dagger.Module;
import dagger.Provides;


@Module
public class SharedPrefModule {

    @Provides
    SharedPrefsUtils provideSharedPrefUtils(@NonNull Context context) {
        return new SharedPrefsUtils(context);
    }
}

File: app/src/main/java/com/tronography/pixelweather/dagger/SharedPrefModule.kt
package com.tronography.pixelweather.dagger

import android.content.Context

import com.tronography.pixelweather.utils.SharedPrefsUtils

import dagger.Module
import dagger.Provides


@Module
class SharedPrefModule {

    @Provides
    internal fun provideSharedPrefUtils(context: Context): SharedPrefsUtils {
        return SharedPrefsUtils(context)
    }
}
